core/src/main/java/org/testcontainers/utility/Versioning.java

```diff
@@ -0,0 +1,66 @@
+package org.testcontainers.utility;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+
+/**
+ * Represents mechanisms for versioning docker images.
+ */
+interface Versioning {
+    boolean isValid();
+
+    String getSeparator();
+
+    @EqualsAndHashCode
+    class TagVersioning implements Versioning {
+        public static final String TAG_REGEX = "[\\w][\\w.\\-]{0,127}";
+        private final String tag;
+
+        TagVersioning(String tag) {
+            this.tag = tag;
+        }
+
+        @Override
+        public boolean isValid() {
+            return tag.matches(TAG_REGEX);
+        }
+
+        @Override
+        public String getSeparator() {
+            return ":";
+        }
+
+        @Override
+        public String toString() {
+            return tag;
+        }
+
+        static final TagVersioning LATEST = new TagVersioning("latest");
```

Comment:
nit: should we move it before the non-static class members?

Reply:
Fixed